src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -57,6 +57,11 @@ def map_pair {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : pair W X ⟶ pair Y Z :
 @[simp] lemma map_pair_left {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : (map_pair f g).app walking_pair.left = f := rfl
 @[simp] lemma map_pair_right {W X Y Z : C} (f : W ⟶ Y) (g : X ⟶ Z) : (map_pair f g).app walking_pair.right = g := rfl
 
+/-- Every functor indexing a (co)product is naturally isomorphic (actually, equal) to a `pair` -/
+def pair_of_diagram {F : discrete walking_pair ⥤ C} :
```

Comment:
I think we should try to stick closer to the mathlib naming schemes. I would prefer `diagram_iso_pair`, I think, and similarly for the relatives below.

Reply:
Thanks. I wasn't happy with the name either, but I failed to come up with anything better. I like your suggestion.

Reply:
@TwoFX, when a conversation doesn't require any further review from the reviewer, would you mind clicking "Resolve conversation"? It helps when the reviewer wants to quickly check whether everything is done!